To: vim_dev@googlegroups.com
Subject: Patch 7.3.389
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.389
Problem:    After typing at a prompt the "MORE" message appears too soon.
Solution:   reset lines_left in msg_end_prompt(). (Eswald)
Files:	    src/message.c


*** ../vim-7.3.388/src/message.c	2011-09-14 15:39:26.000000000 +0200
--- src/message.c	2011-12-30 14:08:05.000000000 +0100
***************
*** 85,91 ****
   *		    need_wait_return to be set.  This is a hack to make ":ts"
   *		    work without an extra prompt.
   * lines_left	    Number of lines available for messages before the
!  *		    more-prompt is to be given.
   * need_wait_return TRUE when the hit-return prompt is needed.
   *		    Reset: After giving the hit-return prompt, when the user
   *		    has answered some other prompt.
--- 85,91 ----
   *		    need_wait_return to be set.  This is a hack to make ":ts"
   *		    work without an extra prompt.
   * lines_left	    Number of lines available for messages before the
!  *		    more-prompt is to be given.  -1 when not set.
   * need_wait_return TRUE when the hit-return prompt is needed.
   *		    Reset: After giving the hit-return prompt, when the user
   *		    has answered some other prompt.
***************
*** 855,860 ****
--- 855,861 ----
      cmdline_row = msg_row;
      msg_col = 0;
      msg_clr_eos();
+     lines_left = -1;
  }
  #endif
  
*** ../vim-7.3.388/src/version.c	2011-12-30 13:39:04.000000000 +0100
--- src/version.c	2011-12-30 14:11:00.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     389,
  /**/

-- 
Proofread carefully to see if you any words out.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
